### ISSUE TITLE
li do not close

### ISSUE THREAD

**Issue by heckmac** [evbacher/gd2md-html#3]
All <li> tags do not have a </li> closure tag.

**Comment by evbacher** [comment]
Thanks for the bug: There is some funkiness about closing nested list items in HTML. For plain lists, we're not using the closing tag, but they're not required in HTML5. If you notice a problem with rendering the generated HTML, please share a doc (view-only permission) that illustrates the problem.

**Comment by romanruns** [comment]
I didn't even know about html-tidy.org 😓
Glad I started following this repo just a few hrs ago.. thanks @evbacher!

**Comment by evbacher** [comment]
Yeah, I suppose HTML Tidy would help clean up some lists that are not closed as well.

**Comment by plocket** [comment]
I'm trying to use this to convert into a docusaurus v2 documentation website. Open `<li>`s in the header from you folks was the first error I ran into, so I figure that at least is relevant in here. Also, I selected markdown, so I was surprised to see the result was mostly html. Most of the content is in tables, though, so that probably caused problems. Regardless, I got a lot of errors, so maybe this will be useful to you folks.

The doc:
https://docs.google.com/document/d/1eDzu9IsSK7uhHHCpniPwVg-hq1EgusznE_aueZUtTsg/edit?usp=sharing

The bugs that keep the codebase from building:
1. The `See top comment block for details on ERRORs and WARNINGs.` list items aren't closed
2. `<hr>` instead of `<hr/>`
3. `<MDXLayout>` isn't closed
4. The 'Below the outline' paragraph `<p>` isn't closed.
5. 'Docassemble is our app development platform' came out `>Docassemble is our app development platform` <- that's from the very beginning of the line.

Maybe tables are still a future goal for this app or maybe I've done something else strange. I'm impressed with how well this tool handled what was there. There are a bunch of other issues, but I can't untangle them, so I'll take this generated code and re-work it by hand. It means I won't be able to report back on more issues. Thanks fo a head start on the content.

**Comment by evbacher** [comment]
Thanks for using this add-on -- I'm sorry it doesn't do everything perfectly (the lament of all converter creators, I suspect :)

Notes on your observations:

All tables get converted to HTML. Markdown tables won't handle most complex table content.

For your other questions:
1. Unclosed lists are the original subject of this bug. While HTML5 does not require lists to be closed, I know that many linters prefer it. I may be able to spend some time on this, but no promises!
2. &lt;hr> is the actual tag according to Mozilla: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr. I'm loathe to change it, even though &lt;hr/> is also accepted.
3. Is the unclosed paragraph causing issues? Again, I don't believe HTML5 requires the closing of paragraphs.
4. I'm not sure what you mean here: is there an extra space at the end?

**Comment by plocket** [comment]
Thanks for taking a look and no worries. There's not enough time in all the world for perfect.

3. All I know is that's the error I got. Not saying you have to do anything about it, just letting you know in case it turned out that you do want to do something. No worries.
4. There is a closing angle bracket at the beginning of the line that wasn't in the text from what I recall.

**Comment by evbacher** [comment]
I believe this is fixed with recent changes. Please let us know if you see evidence of this bug still happening.
